File: src/commonTest/kotlin/com/scoreo/infrastructure/InMemoryPlayerRepository.kt
package com.scoreo.infrastructure

import com.scoreo.domain.model.Player
import com.scoreo.domain.port.PlayerRepository

class InMemoryPlayerRepository : PlayerRepository {
    private val players = mutableListOf<Player>()

    override fun getAll(includeInactive: Boolean): List<Player> =
        if (includeInactive) players.toList() else players.filter { it.active }

    override fun save(player: Player) {
        val idx = players.indexOfFirst { it.id == player.id }
        if (idx >= 0) players[idx] = player else players.add(player)
    }

    override fun saveAll(players: List<Player>) {
        players.forEach { save(it) }
    }

    override fun delete(id: String, anonymize: Boolean) {
        val idx = players.indexOfFirst { it.id == id }
        if (idx == -1) return
        players[idx] = players[idx].copy(
            active = false,
            name = if (anonymize) "" else players[idx].name,
        )
    }
}

File: src/infrastructure/testing/inMemoryPlayerRepository.ts
import type { Player } from '../../domain/model/player'
import type { PlayerRepository } from '../../domain/port/playerRepository'

export class InMemoryPlayerRepository implements PlayerRepository {
  private players: Player[] = []

  getPlayerCount(): number {
    return this.players.length
  }

  getAllPlayers(): Player[] {
    return [...this.players]
  }

  getAll(includeInactive = false): Player[] {
    return includeInactive ? [...this.players] : this.players.filter((p) => p.active)
  }

  save(player: Player): void {
    const idx = this.players.findIndex((p) => p.id === player.id)
    if (idx >= 0) this.players[idx] = player
    else this.players.push(player)
  }

  saveAll(players: Player[]): void {
    players.forEach((p) => this.save(p))
  }

  delete(id: string, anonymize = false): void {
    const idx = this.players.findIndex((p) => p.id === id)
    if (idx === -1) return
    const existing = this.players[idx]
    this.players[idx] = { ...existing, active: false, name: anonymize ? '' : existing.name }
  }

  hardDelete(id: string): void {
    this.players = this.players.filter((p) => p.id !== id)
  }

  deleteAll(): void {
    this.players = []
  }
}
